addons/point_of_sale/models/account_move.py

```diff
@@ -59,6 +60,12 @@ def _compute_payments_widget_reconciled_info(self):
                             'pos_payment_name': pos_payment.payment_method_id.name,
                         })
 
+    def _compute_amount(self):
+        super()._compute_amount()
+        for move in self:
+            if move.move_type == 'entry' and move.reversed_pos_order_id:
+                move.amount_total_signed = move.amount_total_signed * -1
```

Comment:
How is this solving the problem?  As I would suppose that you need it by HSN in the report as well?